### PR TITLE
Fix golint warning on pkg/sysinfo

### PR DESCRIPTION
**Pull request by HuKeping**
Part of #14756
Signed-off-by: Hu Keping hukeping@huawei.com

### CODE REVIEW

File: pkg/sysinfo/sysinfo.go [hunk 1]
```diff
@@ -5,10 +5,10 @@ package sysinfo
 type SysInfo struct {
 	AppArmor bool
```
Comment (runcom) [inline pkg/sysinfo/sysinfo.go]: I think these exported fields should be documented

Comment (HuKeping) [inline pkg/sysinfo/sysinfo.go]: It seems `golint` will not get crazy about that, but I agree with you it is better to add comments on those exported fields.

File: pkg/sysinfo/sysinfo.go [hunk 2]
```diff
@@ -3,23 +3,42 @@ package sysinfo
...
 	*cgroupMemInfo
-	*cgroupCpuInfo
-	IPv4ForwardingDisabled        bool
-	BridgeNfCallIptablesDisabled  bool
-	BridgeNfCallIp6tablesDisabled bool
-	CgroupDevicesEnabled          bool
+	*cgroupCPUInfo
+
+	// If IPv4 forwarding was disabled, networking will not work
```
Comment (vdemeester) [inline pkg/sysinfo/sysinfo.go]: You should choose between "if" and "wether" :stuck_out_tongue_closed_eyes: :wink:.

File: pkg/sysinfo/sysinfo.go [hunk 3]
```diff
@@ -3,23 +3,42 @@ package sysinfo
 // SysInfo stores information about which features a kernel supports.
 // TODO Windows: Factor out platform specific capabilities.
 type SysInfo struct {
+	// Whether support APpArmor or not
```
Comment (vdemeester) [inline pkg/sysinfo/sysinfo.go]: Typo here for ``AppArmor``.

Comment (vdemeester) [inline pkg/sysinfo/sysinfo.go]: For ``SysInfo`` and ``cgroup*Info``, maybe you should add something like : "Wether the current platform supports AppArmor" (or s/platform/kernel/).

**Comment by LK4D4** [comment]
@HuKeping Would you mind to add `pkg/sysinfo` to `hack/make/validate-lint`?

**Comment by HuKeping** [comment]
OK, I'll do that

File: pkg/sysinfo/sysinfo_windows.go [hunk 4]
```diff
@@ -1,6 +1,6 @@
 package sysinfo
 
-// TODO Windows
+// New returns an empty SysInfo for windows by now.
```
Comment (RichardScothern) [inline pkg/sysinfo/sysinfo_windows.go]: by -> for

File: pkg/sysinfo/sysinfo.go [hunk 5]
```diff
@@ -3,23 +3,43 @@ package sysinfo
 // SysInfo stores information about which features a kernel supports.
 // TODO Windows: Factor out platform specific capabilities.
 type SysInfo struct {
+	// Wether the kernel supports AppArmor or not
```
Comment (aaronlehmann) [inline pkg/sysinfo/sysinfo.go]: Typo: should be "Whether"

File: pkg/sysinfo/sysinfo.go [hunk 6]
```diff
@@ -3,23 +3,43 @@ package sysinfo
...
 
-type cgroupCpuInfo struct {
-	CpuCfsPeriod bool
-	CpuCfsQuota  bool
+type cgroupCPUInfo struct {
+	// Whether support CPU CFS(Completely Fair Scheduler) period or not
+	CPUCfsPeriod bool
+
+	// Whether support CPU CFS(Completely Fair Scheduler) quota or not
+	CPUCfsQuota bool
```
Comment (aaronlehmann) [inline pkg/sysinfo/sysinfo.go]: I suggest rewording the comments on struct members in this file to start with the variable each one describes, to be consistent with other godoc comments. Also, "Whether support <x>" should be "Whether <x> is supported".

File: daemon/daemon_unix.go [hunk 7]
```diff
@@ -185,12 +185,12 @@ func (daemon *Daemon) verifyContainerSettings(hostConfig *runconfig.HostConfig,
 	if hostConfig.MemorySwappiness != -1 && (hostConfig.MemorySwappiness < 0 || hostConfig.MemorySwappiness > 100) {
 		return warnings, fmt.Errorf("Invalid value: %d, valid memory swappiness range is 0-100.", hostConfig.MemorySwappiness)
 	}
-	if hostConfig.CPUPeriod > 0 && !daemon.SystemConfig().CpuCfsPeriod {
+	if hostConfig.CPUPeriod > 0 && !daemon.SystemConfig().CPUCfsPeriod {
```
Comment (aaronlehmann) [inline daemon/daemon_unix.go]: CFS is an abbreviation for "completely fair scheduler", so it should be all-caps rather than title case.

File: pkg/sysinfo/sysinfo.go (hunk 6 above)
Comment (HuKeping) [inline pkg/sysinfo/sysinfo.go]: These fields in structure don't like functions that we can  easy to describe them with "FunctionA doing ..." . But if you can give me an example, I'd like to update

File: daemon/daemon_unix.go (hunk 7 above)
Comment (HuKeping) [inline daemon/daemon_unix.go]: It seems we have the similar issue all around the project

**Comment by tiborvass** [comment]
LGTM

**Comment by jeffdm** [comment]
I'm getting the warning:

```
Errors from glint:
pkg/sysinfo/sysinfo_freebsd.go:3:1: comment on exported function New should be of the form "New ..."
```

**Comment by vdemeester** [comment]
@jeffdm this means @HuKeping missed one.
@HuKeping could you update to add it :blush: ? (you can either use ``make validate-lint`` or ``golint pkg/sysinfo/*.go`` to make sure you got all of the files :wink: .

**Comment by HuKeping** [comment]
I'll check that. There is no more warning on my `golint pkg/sysinfo`, maybe it should be `golint pkg/sysinfo/*.go`

**Comment by HuKeping** [comment]
updated, thanks @jeffdm

**Comment by runcom** [comment]
LGTM, waiting windows janky
